src/lib/Util.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity ^0.8.0;
+
+library Util {
+    error NegativeUint();
+    error IndexOutOfBounds();
+    error InvalidArrLength();
+
+    function subIntFromUint(int256 a, uint256 b) public pure returns (uint256) {
```

Comment:
i think you can do operator overloading so this is usable via a `-`, if not then `using for` at least

Reply:
Unfortunately to overload the `-` both values need to be of the same type. I've added a new `sub` for uint256 types.